zio-http-gen/src/main/scala/zio/http/gen/openapi/EndpointGen.scala

```diff
@@ -527,7 +671,8 @@ final case class EndpointGen() {
               throw new Exception(s"Unexpected subtype $other for oneOf schema $schema")
           }
           .toList
-        val noDiscriminator                      = caseNames.isEmpty
+        val noDiscriminator                         = caseNames.isEmpty
+        val abstractMembersFields: List[Code.Field] = abstractMembers.flatMap(fieldsOfObject(openAPI, annotations))
```

Comment:
validate same field does not appear multiple times (with different type; if same type, we can distinct - also, if same type but optional - i.e. one required, and another optional - we can omit optional and stay with required only)

Same treatment should be done for case classes as well